apollo-router/src/configuration/mod.rs

```diff
@@ -629,6 +639,88 @@ pub(crate) struct Supergraph {
     /// Log a message if the client closes the connection before the response is sent.
     /// Default: false.
     pub(crate) experimental_log_on_broken_pipe: bool,
+
+    /// Configuration options pertaining to the query planner component.
+    pub(crate) query_planner: QueryPlanner,
+}
+
+/// Configuration options pertaining to the query planner component.
+#[derive(Debug, Clone, Default, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct QueryPlanner {
+    /// Set the size of a pool of workers to enable query planning parallelism.
+    /// Default: 1.
+    pub(crate) experimental_available_parallelism: AvailableParallelism,
+}
+
+#[derive(Debug, Clone, JsonSchema)]
+#[serde(untagged)]
+pub(crate) enum AvailableParallelism {
```

Comment:
If you want to avoid implementing `Serialize` and `Deserialize` you can do the same trick we did for `HeartbeatInterval` in our codebase

Reply:
great idea! done in [a5209a8](https://github.com/apollographql/router/pull/4897/commits/a5209a8a0518e4fd257559b7efb6a44508bb2f2c)